src/segment_node.go

```diff
@@ -22,13 +35,26 @@ func (n *node) init(props *properties, env environmentInfo) {
 		},
 		versionURLTemplate: "[%[1]s](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V%[2]s.md#%[1]s)",
 		matchesVersionFile: n.matchesVersionFile,
+		loadContext:        n.loadContext,
 	}
 }
 
 func (n *node) enabled() bool {
 	return n.language.enabled()
 }
 
+func (n *node) loadContext() {
+	if !n.language.props.getBool(DisplayPackageManager, false) {
+		return
+	}
+	if n.language.env.hasFiles("yarn.lock") {
+		n.packageManagerIcon = n.language.props.getString(YarnIcon, " \uF61A")
+	}
+	if n.language.env.hasFiles("package-lock.json") || n.language.env.hasFiles("package.json") {
+		n.packageManagerIcon = n.language.props.getString(NPMIcon, " \uE71E")
+	}
```

Comment:
@JanDeDobbeleer should this be an `if/else if`? In my yarn based projects I'm seeing the npm icon and I think it's because they contain a `yarn.lock` and a `package.json` which the second `if` is picking up.

Reply:
You're right. There was a return in here but that got removed somehow 🤦🏻‍♂️

Reply:
Fixed.